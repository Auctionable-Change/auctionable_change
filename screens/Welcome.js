import * as React from "react";
import { View, Button } from "react-native";

const Welcome = ({ navigation }) => {
  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <Button
        title="Current Listings"
        onPress={() => navigation.navigate("Current Listings")}
      />
    </View>
  );
};

export default Welcome;
