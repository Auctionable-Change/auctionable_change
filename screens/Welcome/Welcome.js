import * as React from "react";
import { View, Image } from "react-native";
import NavBar from '../../components/NavBar/NavBar';
import { StyleSheet } from "react-native";
import { Button, Text } from "native-base";


const Welcome = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Image
        source={require("../../assets/stacked_logo.png")}
        style={styles.logo}
      />
      <Text style={styles.motto}>
        Buy and sell with purpose because together we can make A.change
      </Text>
      <View style={{ flex: 1, padding: 10 }}>
        <Button block info onPress={() => navigation.navigate("Listings")}>
          <Text style={{ fontFamily: "quicksand-bold" }}>Current Listings</Text>
        </Button>
        <Button
          style={{ marginTop: 30 }}
          block
          info
          onPress={() => navigation.navigate("Post Item")}
        >
          <Text style={{ fontFamily: "quicksand-bold" }}>Post Item</Text>
        </Button>
      </View>
      <NavBar navigation={navigation} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  logo: {
    width: "95%",
    height: 200,
    resizeMode: "contain",
  },
  motto: {
    width: "70%",
    textAlign: "center",
    alignSelf: "center",
    color: "grey",
    marginBottom: 10,
    fontFamily: "quicksand",
  },
});

export default Welcome;
