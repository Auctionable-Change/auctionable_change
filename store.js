import React, { createContext, useContext, useReducer } from "react";

const StoreContext = createContext();

const initialState = {
  currentListing: {
    id: null,
    title: "",
    description: "",
    price: null,
    donor: "",
    donor_email: "",
    status: "",
    category: "",
    charity: "",
    charity_url: "",
    charity_score: "",
    charity_score_image: "",
    image: "",
    bids: [],
  },
  buyerDetails: {
    item_id: null,
    bidder_name: "",
    bidder_email: "",
    amount: null,
    street_address: "",
    city: "",
    state: "",
    zip_code: "",
    receipt: "",
  },
  listingToPost: {},
};

const reducer = (state, action) => {
  switch (action.type) {
    case "ADD_CURRENT_LISTING":
      return {
        ...state,
        currentListing: action.currentListing,
      };
    case "ADD_SELECTED_CHARITY":
      return {
        ...state,
        listingToPost: {
          ...state.listingToPost,
          charity: action.selectedCharity.name,
          charity_url: action.selectedCharity.url,
          charity_score: action.selectedCharity.rating,
          charity_score_image: action.selectedCharity.rating_image,
        },
      };
    case "ADD_BUYER_DETAILS":
      return {
        ...state,
        buyerDetails: { ...state.buyerDetails, ...action.buyerDetails },
      };
    case "ADD_TO_LISTING":
      return {
        ...state,
        listingToPost: { ...state.listingToPost, ...action.listingToPost },
      };
    case "POST_BID":
      return {
        ...state,
        currentBid: action.bid
      };
      case "SET_USER":
      return {
        ...state,
        loginDetails: action.loginDetails,
        userInfo: action.userInfo,
        registrationDetails: action.registration
      }

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={{ state, dispatch }}>
      {children}
    </StoreContext.Provider>
  );
};

export const useStore = () => useContext(StoreContext);
