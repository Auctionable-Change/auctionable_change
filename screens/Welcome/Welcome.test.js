import React from 'react';
import Welcome from './Welcome';
import { StoreProvider } from '../../store'
import {expect, it} from '@jest/globals';
import renderer from "react-test-renderer";


describe('Welcome', () => {
  it('should render', () => {
    const tree = renderer.create(
      <StoreProvider>
        <Welcome />
      </StoreProvider>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})