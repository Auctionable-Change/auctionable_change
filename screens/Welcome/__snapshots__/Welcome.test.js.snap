// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "resizeMode": "contain",
        "width": "95%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "alignSelf": "center",
          "color": "grey",
          "fontFamily": "quicksand",
          "marginBottom": 10,
          "textAlign": "center",
          "width": "70%",
        },
      ]
    }
  >
    Buy and sell with purpose because together we can make A.change
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#62B1F6",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "fontFamily": "quicksand-bold",
            },
          ]
        }
      >
        Current Listings
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#62B1F6",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "fontFamily": "quicksand-bold",
            },
          ]
        }
      >
        Post Item
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopColor": "#dddddd",
        "borderTopWidth": 1,
        "bottom": "0%",
        "display": "flex",
        "flexDirection": "row",
        "height": "8%",
        "justifyContent": "space-around",
        "padding": 8,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="search"
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="add"
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="home"
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
